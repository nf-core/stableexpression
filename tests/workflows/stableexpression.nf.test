nextflow_workflow {

    name "Test Workflow STABLEEXPRESSION"
    script "workflows/stableexpression.nf"
    workflow "STABLEEXPRESSION"
    config "./nextflow.config"
    tag "workflow"

    test("Two Expression Atlas accessions provided") {

        tag "workflow_eatlas_accessions"

        when {
            params {
                species = "solanum tuberosum"
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Expression Atlas accession - two output datasets") {

        tag "workflow_eatlas_accession_two_datasets"

        when {
            params {
                species = "homo sapiens"
                eatlas_accessions = "E-GEOD-1615"
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Two Expression Atlas no keyword (whole species)") {

        tag "workflow_eatlas_no_kw"

        when {
            params {
                species = "solanum tuberosum"
                fetch_eatlas_accessions = true
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Two Expression Atlas keywords provided") {

        tag "workflow_eatlas_kw"

        when {
            params {
                species = "solanum tuberosum"
                eatlas_keywords = "potato,stress"
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Two Expression Atlas accessions provided - normalisation with EdgeR") {

        tag "workflow_eatlas_accessions_edger"

        when {
            params {
                species = "solanum tuberosum"
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
                normalisation_method = "edger"
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Test workflow - common accession (E-MTAB-552) between manual and auto") {

        tag "workflow_accession_E-MTAB-552"

        when {
            params {
                species = "solanum tuberosum"
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
                eatlas_keywords = "phloem"
            }
            workflow {
                """
                input[0] = Channel.empty()
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}
