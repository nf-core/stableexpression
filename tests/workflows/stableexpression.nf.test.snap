{
    "Expression Atlas accession - two output datasets": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,24892b65e3569872371f79c3afcb863d"
            ],
            [
                "stats_all_genes.csv:md5,7867fdf5168199d20dc16c47006ef8ce"
            ],
            [
                "count_summary.csv:md5,3ce9f092b0863842fc08db3dd0bc947e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T10:46:57.423487611"
    },
    "Two Expression Atlas keywords provided": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,a06dceee03c9d413d6c8ec22329f5262"
            ],
            [
                "stats_all_genes.csv:md5,ef718e06c30989d1e8341cbd2f5b8b44"
            ],
            [
                "count_summary.csv:md5,42fd89bef2f7a1e6f43ae91bc443e584"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T10:55:01.778957099"
    },
    "Two Expression Atlas no keyword (whole species)": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,efb81b25c691f8758da1c1f44f315016"
            ],
            [
                "stats_all_genes.csv:md5,0ec1c7005ad2c0ef43d4b43787293cfc"
            ],
            [
                "count_summary.csv:md5,7bffef35d4d295abd25391ff42def670"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T10:50:56.83921757"
    },
    "Test workflow - common accession (E-MTAB-552) between manual and auto": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,b311ed6e0abab127f180ece1737c4835"
            ],
            [
                "stats_all_genes.csv:md5,a1ee2aa140ec3f2968b662e46e65a7ae"
            ],
            [
                "count_summary.csv:md5,ff830820097999847752e1e553628c18"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T15:21:40.286621572"
    },
    "Two Expression Atlas accessions provided - normalisation with EdgeR": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,9af4528cb8ea2695e0aff66c1ea464a7"
            ],
            [
                "stats_all_genes.csv:md5,7339175b11aeaeac1a9552a294f9f7b3"
            ],
            [
                "count_summary.csv:md5,06ba6597605876cc7152c67db0545075"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T10:56:26.563087788"
    },
    "Two Expression Atlas accessions provided": {
        "content": [
            [
                "stats_most_stable_genes.csv:md5,91a38d6845c7a8bdd744b5bd7d14580d"
            ],
            [
                "stats_all_genes.csv:md5,a209dc78d21f5ec26dadccc33a121602"
            ],
            [
                "count_summary.csv:md5,d663b54f74d651807188ccf1eb5ca7e2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-09T10:46:17.955565042"
    }
}